paddle/fluid/framework/executor.cc

```diff
@@ -321,7 +321,8 @@ std::vector<std::shared_ptr<ExecutorPrepareContext>> Executor::Prepare(
 }
 
 void Executor::RunPreparedContext(ExecutorPrepareContext* ctx, Scope* scope,
-                                  bool create_local_scope, bool create_vars) {
+                                  bool create_local_scope, bool create_vars,
+                                  bool drop_scope_kids) {
```

Comment:
It's not a good way to add a bool switch here. I think a better way is to let `listen_and_serv_op` use a child scope.

Reply:
I'll try remove the `create_vars` bool in executor to see if we can solve this

Reply:
Agree with using a sub-scope in `listen_and_serv_op`.